src/main/scala/vectorpipe/VectorPipe.scala

```diff
@@ -113,12 +113,9 @@ object VectorPipe {
     * @param clip A function which represents a "clipping strategy".
     * @param logError An IO function that will log any clipping failures.
     */
-  def toGrid[G <: Geometry, D](
-    clip: (Extent, Feature[G, D]) => Option[Feature[G, D]],
-    logError: (Extent, Feature[G, D]) => Unit,
-    ld: LayoutDefinition,
-    rdd: RDD[Feature[G, D]]
-  ): RDD[(SpatialKey, Iterable[Feature[G, D]])] = {
+  def toGrid[G <: Geometry, D](ld: LayoutDefinition, rdd: RDD[Feature[G, D]])
+            (clip: (Extent, Feature[G, D]) => Option[Feature[G, D]])
+            (logError: (((Extent, Feature[G, D])) => String) => ((Extent, Feature[G, D])) => Unit): RDD[(SpatialKey, Iterable[Feature[G, D]])] = {
```

Comment:
Multiple arg lists to be more Scala-like.

Reply:
Minor nitpick: it seems to me that the first argument should be the one that changes least in a given program. My intuition here is that different clipping strategies might make sense while different logging behaviors would be somewhat strange. Consider inverting the argument order

Reply:
> the first argument should be the one that changes least in a given program

I follow that philosophy as well. So really the functions should be first, except "functions last" seems to be a Scala thing. I'm caught on what to do.

Reply:
I didn't realize there was best practice around this in the community. So much the worse for scala's standards - I think you should follow your heart and put functions before simpler bits of the ontology

Reply:
https://docs.scala-lang.org/style/declarations.html#multiple-parameter-lists

Multiple args aren't necessarily more scala-like, especially if there are more than one.

The Scala-like separation of a function and the rest of the param list really makes sense when there is usually a lambda being passed in, so that you can get a syntax like this:

```scala
doAThing(gonnaDoItToThis) { x => // the thing that does the thing }
```

This breaks when you have multiple functions that are really just params to the function, and are auxiliary to the main point of the function.

So my 2cents is to remove the multiple params here.

Reply:
It's really doubtful that somebody would implement their custom `clip` as a lambda to `toGrid`. Likewise for the logging functions, so those are used in two places.

Reply:
Although there is a point from the document that @lossyrob just linked:

> For complex DSLs, or with type-names that are long, it can be difficult to fit the entire signature on one line. 

which is true for this case. The code and scaladocs render a bit prettier with multiple arg lists.